frontend/apps/web/app/(mgmt)/[account]/jobs/[id]/source/components/DataSyncConnectionCard.tsx

```diff
@@ -418,14 +432,24 @@ export default function DataSyncConnectionCard({ jobId }: Props): ReactElement {
   async function validateVirtualForeignKeys(
     vfks: VirtualForeignConstraintFormValues[]
   ) {
+    const values = form.getValues();
+    const connection = connections.find((c) => c.id === values.sourceId);
+    const job = data?.job;
     try {
       setIsValidatingMappings(true);
+      let jobsource: JobSource | undefined;
+      if (job && connection) {
+        jobsource = create(JobSourceSchema, {
+          options: toJobSourceOptions(values, job, connection, values.sourceId),
+        });
+      }
```

Comment:
seems liek this logic is pretty similar to what is above in the validateMappings function, but it's not using the same conditional logic. ... should it be? they look pretty darn similar.